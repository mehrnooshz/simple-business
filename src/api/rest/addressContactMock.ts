export interface Props {
  title: string;
  descriptionFirst: string;
  descriptionSecond: string;
}

const addressContactMock: Props = {
  title: 'test title',
  descriptionFirst: 'first description',
  descriptionSecond: 'second description',
};

export default addressContactMock;
