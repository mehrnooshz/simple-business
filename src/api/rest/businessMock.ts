import Business from 'api/model/business';

export const BusinessMock = {
  name: 'test',
  id: '1',
  description: 'hi',
  phone: '0911',
  image: 'url',
  email: 'a@yahoo.com',
  address: {
    number: '021',
    street: ' book',
    zip: '123',
    city: 'rasht',
    country: 'iran',
  },
};
export const BusinessSameMock = {
  name: 'test2',
  id: '2',
  description: 'hi',
  phone: '0911',
  image: 'url2',
  email: 'b@yahoo.com',
  address: {
    number: '011',
    street: 'cover',
    zip: '456',
    city: 'rasht',
    country: 'iran',
  },
};

export const BusinessDifferentMock = {
  name: 'test3',
  id: '3',
  description: 'hi',
  phone: '0911',
  image: 'url2',
  email: 'c@yahoo.com',
  address: {
    number: '011',
    street: 'cover',
    zip: '456',
    city: 'sofia',
    country: 'bulgharia',
  },
};

export const BusinessesMock: Business[] = [
  BusinessMock,
  BusinessSameMock,
  BusinessDifferentMock,
];

export const businessIdExistedMock: string = '1';
export const businessIdNotExistedMock: string = '4';
