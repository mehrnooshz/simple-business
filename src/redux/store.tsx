import {
  applyMiddleware,
  combineReducers,
  createStore,
  Store,
  AnyAction,
} from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import AppState from './state/appState';
import BusinessListReducer from './reducers/businessReducer';

const rootReducer = combineReducers<AppState>({
  businessListState: BusinessListReducer,

  // TODO: add sub-reducers here...
});

export default function configureStore(): Store<AppState, AnyAction> {
  return createStore(
    rootReducer,
    undefined,
    composeWithDevTools(applyMiddleware(thunk))
  );
}
