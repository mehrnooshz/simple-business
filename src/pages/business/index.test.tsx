import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer, {
  ReactTestRenderer,
  ReactTestInstance,
} from 'react-test-renderer';
import { BusinessMock, BusinessSameMock } from 'api/rest/businessMock';
import Business from './';
import AppState from 'redux/state/appState';
import { BrowserRouter } from 'react-router-dom';
import { Store } from 'redux';
import AddressContactContainer from 'components/business/addressContactContainer';
import ImageContainer from 'components/business/imgContainer';
import NearPlaces from 'components/business/nearPlaces';
import { cleanup } from '@testing-library/react';
describe('Business', () => {
  const createMockStore = configureStore<AppState>([thunk]);

  const createEmptyState = (): AppState => {
    return {
      businessListState: {
        businesses: [],
        sameBusinesses: [],
        business: undefined,
        isFetching: false,
      },
    };
  };

  const createTestRenderer = (store: Store<AppState>): ReactTestRenderer => {
    return renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Business />
        </Provider>
      </BrowserRouter>
    );
  };

  describe('should be rendered', () => {
    it('inside a redux Provider', () => {
      createTestRenderer(createMockStore(createEmptyState()));
    });

    it('and it should match the expected snapshot', () => {
      expect(
        createTestRenderer(createMockStore(createEmptyState())).toJSON()
      ).toMatchSnapshot();
    });

    describe('if a business exist', () => {
      const business = BusinessMock;
      const sameBusiness = BusinessSameMock;
      const state = createEmptyState();
      state.businessListState.business = business;
      state.businessListState.sameBusinesses = [sameBusiness];
      state.businessListState.business = business;

      let component: ReactTestRenderer;
      let addressContactContainer: ReactTestInstance;
      let image: ReactTestInstance;
      let nearPlaces: ReactTestInstance;

      beforeEach(() => {
        component = createTestRenderer(createMockStore(state));
        image = component.root.findByType(ImageContainer);
        nearPlaces = component.root.findByType(NearPlaces);
        addressContactContainer = component.root.findByType(
          AddressContactContainer
        );

        const element = component.root.findByType('div');
        expect(element.props.className.includes('container')).toBe(true);
      });

      it('ImageContainer check props', () => {
        expect(image.props.imgSrc).toBe(business.image);
      });

      it('AddressContactContainer check props business', () => {
        expect(addressContactContainer.props.business).toBe(business);
      });

      it('samebusiness existed', () => {
        expect(nearPlaces.props.sameBusinesses).toEqual([sameBusiness]);
      });
      afterEach(() => {
        cleanup();
      });
    });
  });

  describe('if business no exist', () => {
    const business = undefined;
    const state = createEmptyState();
    state.businessListState.business = business;

    const mockStore = configureStore([thunk]);
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={mockStore(state)}>
          <Business />
        </Provider>
      </BrowserRouter>
    );
  });
});
