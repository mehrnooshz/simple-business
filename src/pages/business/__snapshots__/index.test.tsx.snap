// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business should be rendered and it should match the expected snapshot 1`] = `
<div
  className="business-container"
>
  <div
    className="container"
  />
</div>
`;
